### PR TITLE
[scarb-doc] Inner attributes

### PR DESCRIPTION
**Pull request by drknzz**

### CODE REVIEW

File: extensions/scarb-doc/tests/test.rs [hunk 1]
```diff
@@ -91,24 +106,46 @@ fn test_main() {
         .success();
     let stdout = std::str::from_utf8(&output.get_output().stdout).unwrap();
 
+    println!("{:?}", &stdout);
```
Comment (Arcticae) [inline extensions/scarb-doc/tests/test.rs:109]:
```suggestion
```

File: extensions/scarb-doc/src/types.rs [hunk 2]
```diff
@@ -135,22 +138,39 @@ impl Module {
...
+            text_without_trivia: node.lookup(db).as_syntax_node().get_text_without_trivia(db),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct LookupItemData {
+    pub doc: Option<String>,
+    pub definition: String,
+}
```
Comment (Arcticae) [inline extensions/scarb-doc/src/types.rs:167]: Why is a separate struct needed here?

File: extensions/scarb-doc/src/types.rs [hunk 3]
```diff
@@ -214,15 +252,51 @@ pub struct Struct {
...
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Member {
+    pub id: MemberId,
+    pub node: ast::MemberPtr,
+
+    pub item_data: ItemData,
+    // TODO: no LookupItemId enum variant for struct / enum items
```
Comment (Arcticae) [inline extensions/scarb-doc/src/types.rs:289]: I kind of worry that we'll lose the track of those

File: extensions/scarb-doc/src/types.rs [hunk 4]
```diff
@@ -286,15 +408,118 @@ pub struct Trait {
...
+}
+
+#[derive(Clone, Debug)]
+pub struct TraitConstant {
+    pub id: TraitConstantId,
+    pub node: ast::TraitItemConstantPtr,
+
+    pub item_data: ItemData,
+    pub lookup_item_data: LookupItemData,
+}
```
Comment (Arcticae) [inline extensions/scarb-doc/src/types.rs:462]: Why this can't be a regular constant? 🤔

Comment (Arcticae) [reply extensions/scarb-doc/src/types.rs:462]: I guess this applies to all of the nested things inside of the traits

File: extensions/scarb-doc/src/types.rs [hunk 5]
```diff
@@ -304,15 +529,83 @@ pub struct Impl {
...
+
+#[derive(Clone, Debug)]
+pub struct ImplFunction {
+    pub id: ImplFunctionId,
+    pub node: ast::FunctionWithBodyPtr,
+
+    pub item_data: ItemData,
+    pub lookup_item_data: LookupItemData,
+}
```
Comment (Arcticae) [inline extensions/scarb-doc/src/types.rs:596]: This also seems like something that could be generalised by a function type

File: extensions/scarb-doc/src/types.rs (hunk 3 above)
Comment (piotmag769) [reply extensions/scarb-doc/src/types.rs]: Fixed

File: extensions/scarb-doc/src/types.rs (hunk 4 above)
Comment (piotmag769) [reply extensions/scarb-doc/src/types.rs:479]: Cairo stuff I guess

File: extensions/scarb-doc/src/types.rs (hunk 5 above)
Comment (piotmag769) [reply extensions/scarb-doc/src/types.rs:645]: Depends if we want to keep it isomorphic to structs from compiler or not

File: extensions/scarb-doc/src/types.rs (hunk 2 above)
Comment (piotmag769) [reply extensions/scarb-doc/src/types.rs]: Will remove, good point

File: extensions/scarb-doc/src/types.rs (hunk 5 above)
Comment (Arcticae) [reply extensions/scarb-doc/src/types.rs]: What do you think?

Comment (piotmag769) [reply extensions/scarb-doc/src/types.rs]: I would keep it this way for now, if we don't find any use case for that we can merge it

**Comment by piotmag769** [comment]
@Arcticae I will open a separate PR with new database for `scarb-doc`, merge it first into main and then rebase this one on top of main. It is due to Cairo already being bumped on main

File: extensions/scarb-doc/src/types.rs [hunk 6]
```diff
@@ -184,12 +224,19 @@ pub struct Use {
     pub item_data: ItemData,
 }
 
+// TODO: do we even want to document this?
```
Comment (piotmag769) [inline extensions/scarb-doc/src/types.rs:227]: @Arcticae

File: Cargo.toml [hunk 7]
```diff
@@ -136,7 +136,7 @@ zip = { version = "0.6", default-features = false, features = ["deflate"] }
 zstd = "0.13"
 
 [profile.release]
-lto = true
```
Comment (mkaput) [inline Cargo.toml:139]: why

Comment (piotmag769) [reply Cargo.toml:139]: Daily me forgetting about this xD

File: extensions/scarb-doc/src/types.rs (hunk 6 above)
Comment (Arcticae) [reply extensions/scarb-doc/src/types.rs:227]: Not really, wouldn't document it

**Review by Arcticae — APPROVED** [review]
LGTM

**Pull request merged**